tests/posix/common/src/pthread.c

```diff
@@ -371,6 +377,67 @@ ZTEST(pthread, test_pthread_termination)
 	zassert_equal(ret, ESRCH, "cancelled a terminated thread!");
 }
 
+ZTEST(pthread, test_pthread_tryjoin)
+{
+	pthread_t th = {0};
+	void *retval;
+
+	/* Creating a threads that exits after 1s*/
+	zassert_ok(pthread_create(&th, NULL, timedjoin_thread, NULL));
+	usleep(USEC_PER_MSEC * 100U);
+
+	/* Attempting to join immediately should fail */
+	zassert_equal(pthread_tryjoin_np(th, &retval), EBUSY);
+
+	/* Sleep so thread will exit */
+	usleep(USEC_PER_MSEC * 200U);
+
+	/* Attempting to join immediately should succeed now */
```

Comment:
Just thought that this might be clearer (at least for me)
```suggestion
	/* Attempting to join without blocking should succeed now */
```

Reply:
applied. Thanks.